contracts/FlowTransactionSchedulerUtils.cdc

```diff
@@ -0,0 +1,335 @@
+            } else {
+                self.idsByHandler[handlerTypeIdentifier] = [id]
+            }
+
+            return id
+        }
+
+        /// Cancel a scheduled transaction by its ID
+        /// @param id: The ID of the transaction to cancel
+        /// @return: A FlowToken vault containing the refunded fees
+        access(Owner) fun cancel(id: UInt64): @FlowToken.Vault {
+            // Remove the transaction from our dictionary
+            let tx <- self.scheduledTransactions.remove(key: id)
+                ?? panic("Invalid ID: Transaction with ID \(id) not found in manager")
+
+            let transactionData = FlowTransactionScheduler.getTransactionData(id: id)
+                ?? panic("Invalid ID: Transaction with ID \(id) not found in scheduler")
+
+            self.removeID(id: id, timestamp: tx.timestamp, handlerTypeIdentifier: tx.handlerTypeIdentifier)
+
+            // Cancel the transaction through the main scheduler
+            let refundedFees <- FlowTransactionScheduler.cancel(scheduledTx: <-tx!)
+
+            return <-refundedFees
+        }
+
+        /// Remove an ID from the manager's fields
+        /// @param id: The ID of the transaction to remove
+        /// @param timestamp: The timestamp of the transaction to remove
+        /// @param handlerTypeIdentifier: The type identifier of the handler of the transaction to remove
+        access(self) fun removeID(id: UInt64, timestamp: UFix64, handlerTypeIdentifier: String) {
+            if let ids = self.idsByTimestamp[timestamp] {
+                let index = ids.firstIndex(of: id)
+                ids.remove(at: index!)
+                if ids.length == 0 {
+                    self.idsByTimestamp.remove(key: timestamp)
+                } else {
+                    self.idsByTimestamp[timestamp] = ids
+                }
+            }
+
+            if let ids = self.idsByHandler[handlerTypeIdentifier] {
+                let index = ids.firstIndex(of: id)
+                ids.remove(at: index!)
+                self.idsByHandler[handlerTypeIdentifier] = ids
+            }
+        }
+
+        /// Clean up transactions that are no longer valid (return nil or Unknown status)
+        /// This removes and destroys transactions that have been executed, canceled, or are otherwise invalid
+        /// @return: The transactions that were cleaned up (removed from the manager)
+        access(Owner) fun cleanup(): [UInt64] {
+            let currentTimestamp = getCurrentBlock().timestamp
+            var transactionsToRemove: [UInt64] = []
+
+            let pastTimestamps = self.sortedTimestamps.getBefore(current: currentTimestamp)
+            for timestamp in pastTimestamps {
+                let ids = self.idsByTimestamp[timestamp] ?? []
+                for id in ids {
+                    let status = FlowTransactionScheduler.getStatus(id: id)
+                    if status == nil || status == FlowTransactionScheduler.Status.Unknown {
```

Comment:
Do you need to add a cancelled or executed check here?

Reply:
I was thinking that we'd like to keep the executed transaction statuses in the manager for some amount of time after they are executed in case the owner wants to query the status or information about it or something like that, so right now, we're only removing transactions if they no longer have a status. 

It might be better idea to set a limit on the number here because depending on the behavior of the main contract, the storage could start to get a little out of hand. But I think it probably is better to just monitor it for now and we can always add those limits later if needed